### PR TITLE
[PB-3009]: feat/creation modification times on folders

### PR DESCRIPTION
**Pull request by jzunigax2**
Added `creationTime` and `modificationTime` fields to the folders table. When creating a new folder it is allowed to set these parameters optionally.
Also when modifying a folder or file's name the respective `modificationTime` field gets updated.

### CODE REVIEW

File: migrations/20240929040050-create-modificationtime-index-on-files.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+const tableName = 'files';
+const indexName = 'files_modification_time_index';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface) {
```
Comment (sg-gs) [inline migrations/20240929040050-create-modificationtime-index-on-files.js:8]: I need these indexes to be created CONCURRENTLY @jzunigax2. Thus, I can run them in the background without locking the `folders` table. It's less clean but far more effective.

Comment (sg-gs) [reply migrations/20240929040050-create-modificationtime-index-on-files.js:8]: Ref: https://www.postgresql.org/docs/current/sql-createindex.html

File: migrations/20240929044035-create-modificationtime-index-on-folders.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+const tableName = 'folders';
+const indexName = 'folders_modification_time_index';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface) {
+    await queryInterface.addIndex(tableName, ['modification_time'], {
```
Comment (sg-gs) [inline migrations/20240929044035-create-modificationtime-index-on-folders.js:9]: Same here

**Review by apsantiso — APPROVED** [review]
LGTM, let's just wait for the migrations to be added.

**Comment by sg-gs** [comment]
Migrations run @jzunigax2 @apsantiso. Ready to merge.

**Comment by apsantiso** [comment]
@sg-gs @jzunigax2 Just made a small commit to remove duplicated migrations to prevent our locals from breaking.

**Pull request merged**